filebeat/crawler/registrar.go

```diff
@@ -138,12 +143,12 @@ func (r *Registrar) writeRegistry() error {
 	}
 
 	encoder := json.NewEncoder(file)
-	encoder.Encode(r.State)
+	encoder.Encode(r.getFullState())
 
 	// Directly close file because of windows
 	file.Close()
 
-	logp.Info("Registry file updated. %d states written.", len(r.State))
+	logp.Info("Registry file updated. %d states written.", len(r.getFullState()))
```

Comment:
getFullState makes a copy holding a mutex. Consider saving to variable before line 146 and print saved copy here.